code/radar/radar.cpp

```diff
@@ -367,43 +388,25 @@ void HudGaugeRadarStd::plotBlip(blip *b, int *x, int *y)
 	if (b->dist < pos->xyz.z) {
 		rscale = 0.0f;
 	} else {
-		rscale = (float) acosf(pos->xyz.z / b->dist) / PI;		//2.0f;	 
+		rscale = (float) acosf(pos->xyz.z / b->dist) / PI;
 	}
 
-	zdist = fl_sqrt((pos->xyz.x * pos->xyz.x) + (pos->xyz.y * pos->xyz.y));
+	zdist = _hypotf(pos->xyz.x, pos->xyz.y);
 
-	float new_x_dist, clipped_x_dist;
-	float new_y_dist, clipped_y_dist;
+	vec3d new_pos;
 
 	if (zdist < 0.01f)
 	{
-		new_x_dist = 0.0f;
-		new_y_dist = 0.0f;
-	}
-	else
-	{
-		new_x_dist = (pos->xyz.x / zdist) * rscale * (Radar_radius[0]/2.0f);
-		new_y_dist = (pos->xyz.y / zdist) * rscale * (Radar_radius[1]/2.0f);
-
-		// force new_x_dist and new_y_dist to be inside the radar
-
-		float hypotenuse;
-		float max_radius;
-
-		hypotenuse = (float) _hypot(new_x_dist, new_y_dist);
-		max_radius = i2fl(Radar_radius[0] - 5);
-
-		if (hypotenuse >= max_radius)
-		{
-			clipped_x_dist = max_radius * (new_x_dist / hypotenuse);
-			clipped_y_dist = max_radius * (new_y_dist / hypotenuse);
-			new_x_dist = clipped_x_dist;
-			new_y_dist = clipped_y_dist;
-		}
+		new_pos = ZERO_VECTOR;
```

Comment:
For some unfathomable reason, this line breaks MSVC 2015. Apparently, you can only assign ZERO_VECTOR to something as part of its declaration (i.e. vec3d x = ZERO_VECTOR; is fine, the code above breaks things).

The easiest solution would be to initialize new_pos to ZERO_VECTOR and invert the zdist check.